### PR TITLE
Malik/feat/auth

### PR DESCRIPTION
**Pull request by malbaker**
- Implemented user auth using `@auth0/auth0-react` SDK
- Created Auth0 HOC to provide Auth0 context
- User is navigated to `/Dashboard` after successful login
- Added route protection to all defined page routes
- Renamed route `/UploadArticles` to `/Upload` to match navbar link
- Defined `/*` route to 404 undefined routes

### CODE REVIEW

File: frontend/src/index.js
Comment (Zachary1575) [inline frontend/src/index.js:1]: I know this sounds very nitpicky, but the spacing is very wide!

**Review by Zachary1575 — CHANGES_REQUESTED** [review]
I know this is abit nitpicky, but the spacing is off for all the files!

**Comment by malbaker** [comment]
I didn't notice the formatting changes right away. I have my own formatting setup in my editor, I didn't think it would cause conflicts. Would you like me to modify my local formatting? I could also help setup a formatting config for the repo to reduce any futures issues around this. @Zachary1575 @am5815

**Comment by Zachary1575** [comment]
What does the formatting configuration look like if applied to the repo? I think as long it maintains readability then it is fine :)

**Comment by malbaker** [comment]
I made an automated format/lint setup in another React project I worked on before, it just needs a couple npm packages. I'll check that out and get back to you with a solution. But if the tab spacing was the major issue (I'm also seeing double vs single quotes for strings), I can just make another PR with the formatting setup implemented

**Pull request merged**